ipv8/gossip/community.py

```diff
@@ -0,0 +1,504 @@
+from abc import abstractmethod
+from random import choice, randint
+
+from twisted.internet import reactor
+from twisted.internet.base import DelayedCall
+from twisted.internet.task import LoopingCall
+
+from ..peer import Peer
+from ..deprecated.bloomfilter import BloomFilter
+from ..keyvault.public.libnaclkey import LibNaCLPK
+from ..messaging.serialization import PackError, Serializable
+from ..overlay import Overlay
+
+# The time required to gather all the necessary votes for carrying out a vote
+VOTE_TIMEOUT = 5
+
+
+class IGossipOverlayListener(object):
+
+    @abstractmethod
+    def on_gossip(self, public_key, message):
+        """
+        Hook method which is called in certain contexts for certain Gossip Messages
+
+        :param public_key: the target_public_key of the message
+        :param message: the message itself
+        :return: None
+        """
+        pass
+
+
+class GossipRule(object):
+    SUPPRESS = 0
+    DEFAULT = 1
+    COLLECT = 2
+    SPREAD = 3
+
+
+class GossipMessage(Serializable):
+    format_list = ['64s', '74s', 'c', '74s', 'raw']
+
+    def __init__(self, my_private_key, rule, target_public_key, payload):
+        self.public_key = my_private_key.pub().key_to_bin()
+        self.payload = payload
+        self.rule = rule
+        self.target_public_key = target_public_key
+
+        self.signature = my_private_key.signature(self.public_key + str(self.rule) + self.target_public_key +
+                                                  self.payload)
+
+    def to_pack_list(self):
+        return [
+            ('64s', self.signature),
+            ('74s', self.public_key),
+            ('c', str(self.rule)),
```

Comment:
Why not make this a `B`? Then you can just put in the `int`, instead of casting the rule to `str` (which is quite fragile and will break horribly if you go over `9`).